src/material/dialog/dialog-container.ts

```diff
@@ -107,6 +89,19 @@ export class MatDialogContainer extends BasePortalOutlet {
     this._document = _document;
   }
 
+  /** Starts the dialog exit animation. */
+  abstract _startExitAnimation(): void;
+
+  /** If the dialog view completes initialization, the open animation starts. */
+  ngAfterViewInit() {
```

Comment:
For context: Previously the previously focused element has been stored before the content has been attached. Additionally the logic in `savePreviouslyFocusedElement` does actually more than it suggests. It also queues a microtask for focusing the dialog container.

The problem here is that in MDC, the dialog can open instantly (if animations are disabled) as we don't go through Angular animations and CD. So we manually start the animation once the view has been initialized. Problem here is now that the `savePreviouslyFocusedElement` method has queued a task for focusing the dialog container... so that will actually override the focus that is set by `_trapFocus()`.

The solution for this is simple: Saving and focusing the dialog container should be done once the dialog content is attached (which is before `ngAfterViewInit` runs). This makes this logic more robust without needing any asynchronous deferring, and fixes the issue in MDC-based dialog container.